lib/src/clojure_lsp/feature/add_missing_libspec.clj

```diff
@@ -269,32 +279,35 @@
                       missing-requires))))))
 
 (defn find-require-suggestions [zloc missing-requires db]
-  (when-let [sexpr (z/sexpr zloc)]
-    (if-let [alias-str (namespace sexpr)]
+  (when-let [sym (safe-sym zloc)]
+    (if-let [alias-str (namespace sym)]
       (find-alias-require-suggestions alias-str missing-requires db)
-      (find-refer-require-suggestions sexpr missing-requires db))))
+      (find-refer-require-suggestions sym missing-requires db))))
 
 (defn add-require-suggestion [zloc chosen-ns chosen-alias chosen-refer db]
-  (->> (find-require-suggestions zloc [] db)
-       (filter #(and (or (= chosen-alias (str (:alias %)))
-                         (= chosen-refer (str (:refer %))))
-                     (= chosen-ns (str (:ns %)))))
-       (map (fn [{:keys [ns alias refer]}]
-              (let [ns-usages-nodes (edit/find-forms zloc #(and (= :token (z/tag %))
-                                                                (symbol? (z/sexpr %))
-                                                                (= ns (-> % z/sexpr namespace))))
-                    known-require (if alias
-                                    (add-known-alias zloc (symbol alias) (symbol ns) db)
-                                    (add-known-refer zloc (symbol refer) (symbol ns) db))]
-                (concat known-require
-                        (when alias
-                          (->> ns-usages-nodes
-                               (map (fn [node]
-                                      (z/replace node (-> (str alias "/" (-> node z/sexpr name))
-                                                          symbol
-                                                          n/token-node
-                                                          (with-meta (meta (z/node  node)))))))
-                               (map (fn [loc]
-                                      {:range (meta (z/node loc))
-                                       :loc loc}))))))))
-       flatten))
+  (let [ns-usages-nodes (edit/find-forms zloc #(when-let [sym (safe-sym %)]
+                                                 (and (= chosen-ns (namespace sym))
+                                                      ;; See note below. This is always false, except in tests
+                                                      (not= chosen-alias (namespace sym)))))
+        chosen-require  (if chosen-alias
+                          (add-known-alias zloc (symbol chosen-alias) (symbol chosen-ns) db)
+                          (add-known-refer zloc (symbol chosen-refer) (symbol chosen-ns) db))]
+    (seq
+      (concat chosen-require
+             ;; TODO: the intention of the following code appears to be to
+             ;; convert un-aliased symbols to aliased symbols, e.g.
+             ;; `clojure.string/split` -> `my-str/split`. However, when on a
+             ;; `clojure.string/split` node, the UI will only ever suggest
```

Comment:
It's possible to pass the alias if using the command manually, not via code action right?

Reply:
Yes, that's true. I realized that too, after I worked on this code. But I forgot to remove this TODO. I'll plan to remove it in a future PR.